packages/core-microservice/src/index.js

```diff
@@ -147,29 +156,31 @@ class Microservice {
           host
         };
         const socket = net.createConnection(opt);
-        socket.on("ready", function (err) {
+        socket.on("ready", (err) => {
           if (err) reject(err);
           resolve();
           socket.destroy();
           isOut = true;
         });
-        socket.on("error", function (err) {
+        socket.on("error", (err) => {
           socket.destroy();
           timeout -= 200;
           if (err.code !== "ECONNREFUSED") {
             isOut = true;
             return reject(err);
           }
           if (timeout > 0) {
-            idTimeout = setTimeout(checker, 200);
+            idTimeout = setTimeout(() => {
+              checker();
+            }, 200);
           } else {
             isOut = true;
             reject(
               new Error(
                 format(
                   Locale().service.run.error_port_timeout,
                   port,
-                  DEFAULT_TIMEOUT
+                  this.timeout
```

Comment:
Could you just double-check that in `this` context, this doesn't refer to something else? But maybe you decide to use a fat arrow instead of a function declaration.